sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -521,19 +521,30 @@ def _get_bigquery_client(project: Optional[str] = None):
  all the logic as the field to GROUP BY the data
 */
 WITH entity_dataframe AS (
-    SELECT
-        *,
-        CONCAT(
-            {% for entity_key in unique_entity_keys %}
-                CAST({{entity_key}} AS STRING),
-            {% endfor %}
-            CAST({{entity_df_event_timestamp_col}} AS STRING)
-        ) AS entity_row_unique_id
+    SELECT *,
+        {{entity_df_event_timestamp_col}} AS entity_timestamp,
+        {% for featureview in featureviews %}
+            CONCAT(
+                {% for entity in featureview.entities %}
+                    CAST({{entity}} AS STRING),
+                {% endfor %}
+                CAST({{entity_df_event_timestamp_col}} AS STRING)
+            ) AS {{featureview.name}}__entity_row_unique_id,
+        {% endfor %}
     FROM {{ left_table_query_string }}
 ),
 
 {% for featureview in featureviews %}
 
+{{ featureview.name }}__entity_dataframe AS (
```

Comment:
Therefore if there are duplicates for a given entity on a given timestamp, this will be drop here to avoid skewed join